tests/test_declaration.py

```diff
@@ -523,6 +535,11 @@ def test_dict(StyleClass):
         ]
     )
 
+    # Properties that are set are in the keys, and unset or invalid ones aren't.
+    assert "explicit_const" in style
+    assert "implicit" not in style
```

Comment:
This is an interesting case... Sure, `implicit` hasn't been defined... but it *does* exist, and it *does* have a value if you try to retrieve it. I'd be inclined to say `implicit in style` is True, not False.

I guess the interesting cases missing in this test are `explicit_none` and `thing`. Is the `in` operator reporting if the style parameter *exists*, or if the style parameter has been *explicitly set*?

Reply:
From a usability standpoint, I feel like being able to check if it's set is more useful. Presumably you define your style class once, so I would expect that 99% of the time you're interested in testing what keys are "in" a given style object, it's the ones currently set on that object. This is consistent with what the `keys` method already did.

That said, I hadn't actually considered how weird directional properties are. If you set the top padding, one would then expect`"padding" in style` to be true... even though it's not actually a "key" on its own.

Reply:
What are you thinking of about the `explicit_none` case that's different from `implicit`?

Reply:
I've made the `in` tests more comprehensive; I've also, at least for now, kept its "only properties that are set" behavior, but expanded it to return `True` for a directional property if at least one of its aliased properties is set.

Reply:
That's a fair point - "have I set this value" is more likely to be the question asked than "is this key valid" (not that either seem especially likely). Plus, I guess we have `hasattr()` to confirm attribute existence.

As for the `explicit_none` case - I guess there isn't really a difference with `implicit` (unless there's a distinction between "has a value at all" and "has had a value explicitly set", which is admittedly even more nebulous than the "property exists" vs "has been set" distinction). I was only drawing attention to it as an exemplar of whether `in` is confirming whether the attribute has a value, or whether the value has been set, but isn't dependent on implicit defaults.

Reply:
Hm... If I'm not mistaken, once they're constructed in the class definition, there's no functional difference between the `implicit` and `explicit_none` properties. Neither of them sets an underscore attribute on a style instance unless a value is assigned to it, and neither initially counts as being "set"; a property can never "actually be" `None`, so that it unambiguously means "not set, with no initial to fall back to".

Reply:
...There are too many kinds of nothing. 😵‍💫